x-pack/plugins/security_solution/public/resolver/store/selectors.ts

```diff
@@ -382,7 +399,8 @@ export const newIDsToRequest: (
 ) => (time: number) => Set<string> = createSelector(
   composeSelectors(dataStateSelector, (dataState: DataState) => dataState.nodeData),
   visibleNodesAndEdgeLines,
-  function (nodeData, visibleNodesAndEdgeLinesAtTime) {
+  composeSelectors(dataStateSelector, dataSelectors.nodeDataIsStale),
+  function (nodeData, visibleNodesAndEdgeLinesAtTime, shouldUpdateAllNodes) {
```

Comment:
Question about when the user clicks refresh or changes the time range filters:

Let's say a user pans around for a while. After that has happened the `nodeData` map will have a bunch of lifecycle events in it for each of the nodes that came into view. Now the user clicks refresh or changes the time range filters. When the node data middleware calls the selector for the nodes in view that selector will return all the nodes because the data for those nodes will be stale. Once that request completes and updates the node data's request ID `shouldUpdateAll` will be false because the request IDs will match.

I think what that means is when the user pans and an "old" node comes into view after the request IDs match, it will not request the data again. I think we can still have stale data in the map.

I'm not sure if this is a bad idea but when the app gets `appReceivedNewExternalProperties` could we wipe out the node data map? That might be ugly but, I don't think there's anything in the map we'd want to keep around. Maybe there's a chance we could get in a weird state though 🤷 

Another ideas is we could move `nodeDataRequestID` inside of the map so each value had one. That way when this selector is called we'd access `nodeData` check to see if that particular node is stale and if so add it to the set of IDs for the middleware to request.